File: backend/internal/models/invoice.go
package models

import (
	"time"
)

type Invoice struct {
	Base
	UserID        string    `json:"user_id"`
	Name          string    `json:"name"`
	Number        *int      `json:"number"`
	Amount        float64   `json:"amount"`
	PaymentMethod string    `json:"payment_method"`
	Description   *string   `json:"description"`
	ItemsQuantity *int      `json:"items_quantity"`
	CustomerName  string    `json:"customer_name"`
	CustomerPhone *string   `json:"customer_phone"`
	CustomerAddr  *string   `json:"customer_address"`
	CustomerEmail *string   `json:"customer_email"`
	Date          time.Time `json:"date"`
}

File: frontend/src/types/invoice.ts
export interface Invoice {
  ID: string;
  number: number;
  name: string;
  amount: number;
  date: string;
  payment_method: string;
  description?: string;
  items_quantity?: number;
  customer_name: string;
  customer_phone?: string;
  customer_address?: string;
  customer_email?: string;
  created_at: string;
  updated_at: string;
}
